fpnge.cc

```diff
@@ -1201,6 +1221,17 @@ extern "C" size_t FPNGEEncode(size_t bytes_per_channel, size_t num_channels,
                          ? (SIMD_WIDTH - (intptr_t)aligned_buf_ptr % SIMD_WIDTH)
                          : 0;
 
+#ifndef FPNGE_FIXED_PREDICTOR
+  std::vector<unsigned char> pdata_buf(bytes_per_line_buf);
+  unsigned char *aligned_pdata_ptr = pdata_buf.data();
+  aligned_pdata_ptr +=
+      (intptr_t)aligned_pdata_ptr % SIMD_WIDTH
+          ? (SIMD_WIDTH - (intptr_t)aligned_pdata_ptr % SIMD_WIDTH)
```

Comment:
A bit orthogonal, but maybe it's worth to factor out these aligned allocation functions to some other class...

(maybe we could also use malloc/new directly, so that no initialization takes place, but IDK)

Reply:
Sounds good, although I'd do it in a future CL.